packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -2457,6 +2481,76 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
     );
   }
 
+  /// This method calculates a local [SelectedContentRange] based on the list 
+  /// of [ranges] that are accumulated from the [Selectable] children under this
+  /// delegate. This calculation takes into account the accumulated content
+  /// length before the active selection, and returns a [SelectedContentRange.empty]
+  /// when either selection edge has not been set.
+  SelectedContentRange _calculateLocalRange(List<SelectedContentRange> ranges) {
+    // Calculate content length.
+    int totalContentLength = 0;
+    for (final SelectedContentRange range in ranges) {
+      totalContentLength += range.contentLength;
+    }
+    if (currentSelectionStartIndex == -1 || currentSelectionEndIndex == -1) {
+      return SelectedContentRange.empty(contentLength: totalContentLength);
+    }
+    int startOffset = 0;
+    int endOffset = 0;
+    bool foundStart = false;
+    bool forwardSelection = currentSelectionEndIndex >= currentSelectionStartIndex;
+    if (currentSelectionEndIndex == currentSelectionStartIndex) {
+      // Determining selection direction is innacurate if currentSelectionStartIndex == currentSelectionEndIndex.
+      // Use the range from the selectable within the selection as the source of truth for selection direction.
+      final SelectedContentRange rangeAtSelectableInSelection = selectables[currentSelectionStartIndex].getSelection();
+      forwardSelection = rangeAtSelectableInSelection.endOffset >= rangeAtSelectableInSelection.startOffset;
+    }
+    for (int index = 0; index < ranges.length; index++) {
```

Comment:
I think my question is still unanswered here: what happens if `foundStart` is always false? You're supposed to return a `SelectedContentRange.empty` right?

Reply:
Sorry missed this one. It should never be `false`, since above it we already check if a `start` and `end` edge exist by checking `currentSelectionStartIndex` and `currentSelectionEndIndex`. If it is `false`, then something is probably wrong. Maybe we should assert instead?

Reply:
Yeah an assert would also make it clear that this is the assumption we are making.